tests/models/nn/test_sasrec.py

```diff
@@ -0,0 +1,1015 @@
+                "cuda",
+                marks=pytest.mark.skipif(torch.cuda.is_available() is False, reason="GPU is not available"),
+            ),
+            pytest.param(
+                "gpu",
+                [0, 1],
+                "cpu",
+                marks=pytest.mark.skipif(
+                    torch.cuda.is_available() is False or torch.cuda.device_count() < 2,
+                    reason="GPU is not available or there is only one gpu device",
+                ),
+            ),
+        ],
+    )
+    @pytest.mark.parametrize(
+        "filter_viewed,expected_cpu_1,expected_cpu_2,expected_gpu",
+        (
+            (
+                True,
+                pd.DataFrame(
+                    {
+                        Columns.User: [30, 30, 40, 40],
+                        Columns.Item: [12, 14, 12, 13],
+                        Columns.Rank: [1, 2, 1, 2],
+                    }
+                ),
+                pd.DataFrame(
+                    {
+                        Columns.User: [30, 30, 40, 40],
+                        Columns.Item: [14, 12, 13, 12],
+                        Columns.Rank: [1, 2, 1, 2],
+                    }
+                ),
+                pd.DataFrame(
+                    {
+                        Columns.User: [30, 30, 40, 40],
+                        Columns.Item: [12, 14, 12, 13],
+                        Columns.Rank: [1, 2, 1, 2],
+                    }
+                ),
+            ),
+            (
+                False,
+                pd.DataFrame(
+                    {
+                        Columns.User: [10, 10, 10, 30, 30, 30, 40, 40, 40],
+                        Columns.Item: [13, 12, 11, 11, 12, 14, 14, 11, 12],
+                        Columns.Rank: [1, 2, 3, 1, 2, 3, 1, 2, 3],
+                    }
+                ),
+                pd.DataFrame(
+                    {
+                        Columns.User: [10, 10, 10, 30, 30, 30, 40, 40, 40],
+                        Columns.Item: [13, 14, 11, 11, 14, 12, 14, 11, 13],
+                        Columns.Rank: [1, 2, 3, 1, 2, 3, 1, 2, 3],
+                    }
+                ),
+                pd.DataFrame(
+                    {
+                        Columns.User: [10, 10, 10, 30, 30, 30, 40, 40, 40],
+                        Columns.Item: [12, 13, 11, 11, 12, 14, 12, 14, 11],
+                        Columns.Rank: [1, 2, 3, 1, 2, 3, 1, 2, 3],
+                    }
+                ),
+            ),
+        ),
+    )
+    def test_u2i(
+        self,
+        dataset_devices: Dataset,
+        filter_viewed: bool,
+        accelerator: str,
+        devices: tp.Union[int, tp.List[int]],
+        recommend_device: str,
+        expected_cpu_1: pd.DataFrame,
+        expected_cpu_2: pd.DataFrame,
+        expected_gpu: pd.DataFrame,
+        recommend_use_torch_ranking: bool,
+    ) -> None:
+
+        if devices != 1:
+            pytest.skip("DEBUG: skipping multi-device tests")
```

Comment:
Multi-device tests work. But.
When multi-device training is enabled, all tests run on all of the devices. This means tests are running multiple times.
Github just kills the whole thing.